src/langchain_google_alloydb_pg/utils/pgvector_migrator.py

```diff
@@ -0,0 +1,440 @@
+    vector_store: AsyncAlloyDBVectorStore,
+    data: Sequence[RowMapping],
+) -> None:
+    """
+    Create a batch insert SQL query to insert multiple rows at once.
+    Looks like
+    INSERT INTO MyTable ( Column1, Column2 )
+    VALUES ( Value1, Value2 ), ( Value1, Value2 ), ...
+
+    Args:
+        vector_store (AlloyDBVectorStore): The AlloyDB vector store corresponding to the table.
+        engine (AlloyDBEngine): The AlloyDB engine corresponding to the Database.
+        destination_table (str): The name of the table to insert the data in.
+        metadata_column_names (str): The metadata columns to be created to keep the data in a row-column format.
+            Optional.
+        data (Sequence[RowMapping]): All the data (to be inserted) belonging to a pgvector collection.
+        use_json_metadata (bool): An option to keep the PGVector metadata as json in the AlloyDB table.
+            Default: False. Optional.
+    """
+    metadata_col_names = (
+        ", " + ", ".join(vector_store.metadata_columns)
+        if len(vector_store.metadata_columns) > 0
+        else ""
+    )
+    insert_stmt = f'INSERT INTO "{vector_store.schema_name}"."{vector_store.table_name}"({vector_store.id_column}, {vector_store.content_column}, {vector_store.embedding_column}{metadata_col_names}'
+    insert_stmt += (
+        f", {vector_store.metadata_json_column})"
+        if vector_store.metadata_json_column
+        else ")"
+    )
+
+    values_stmt = "VALUES (:id, :content, :embedding"
+    for metadata_column in vector_store.metadata_columns:
+        values_stmt += f", :{metadata_column}"
+    if vector_store.metadata_json_column:
+        values_stmt += f", :extra_metadata"
+    values_stmt += ")"
+
+    params = []
+    for row in data:
+        curr_value = {
+            "id": row.id,
+            "content": row.document,
+            "embedding": row.embedding,
+        }
+        metadata = row.cmetadata
+        extra_metadata = row.cmetadata
+        for metadata_column in vector_store.metadata_columns:
+            curr_value[metadata_column] = (
+                metadata[metadata_column] if metadata_column in metadata else None
+            )
+            del extra_metadata[metadata_column]
+        if vector_store.metadata_json_column:
+            curr_value["extra_metadata"] = (
+                json.dumps(extra_metadata) if extra_metadata else None
+            )
+        params.append(curr_value)
+
+    # Insert rows
+    insert_query = insert_stmt + values_stmt
+    async with engine._pool.connect() as conn:
+        await conn.execute(text(insert_query), params)
+        await conn.commit()
+
+
+async def _batch_data(generator, batch_size):
+    """
+    Batches data from a generator.
+
+    Args:
+        generator: A generator function yielding data items.
+        batch_size: The desired batch size.
+
+    Yields:
+        Batches of data as lists.
+    """
+    batch = []
+    async for item in generator:
```

Comment:
This logic may not be needed if we use `fetchmany(size=batch_size)` to manage the batching in the `_aextract_pgvector_collection` method.